### ISSUE TITLE
Document NUGET_CONCURRENCY_LIMIT environment variable

### ISSUE THREAD

**Issue by zivkan** [NuGet/Home#13568]
Needs to be added to the environment variable list table: https://learn.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-environment-variables

PR adding the environment variable: https://github.com/NuGet/NuGet.Client/pull/5046

**Comment by donnie-msft** [comment]
@zivkan since the Release Notes tool collected this issue, I wanted to verify it's not something that's expected to be part of 6.6.x. Is that correct?

**Comment by zivkan** [comment]
no, documentation doesn't ship in the product (and this issue also hasn't been completed)

**Comment by donnie-msft** [comment]
> no, documentation doesn't ship in the product (and this issue also hasn't been completed)

That's correct, docs don't ship. However, I think the tool calls it out since the source change was made, and there's a linked, active issue for the corresponding docs. Thanks for confirming this issue can be ignored.
